src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+                                .collect(Collectors.toList());
+
+                        AlignmentRegion current = alignmentRegionList.get(0);
+                        AlignmentRegion previous;
+                        for (int i = 0; i < alignmentRegionList.size(); i++) {
+                            if (i > 0) {
+                                previous = current;
+                                current = alignmentRegionList.get(i);
+                                final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint();
+                                assembledBreakpoint.contigId = contigId;
+
+                                assembledBreakpoint.region1 = previous;
+                                assembledBreakpoint.region2 = current;
+
+                                assembledBreakpoint.homology = "NA";
+                                if (assembledBreakpoint.region1.endInContig >= assembledBreakpoint.region2.startInContig) {
+                                    assembledBreakpoint.homology = getSequence(contigsCollection, contigId, assembledBreakpoint.region2.startInContig - 1, assembledBreakpoint.region1.endInContig);
+                                }
+
+                                assembledBreakpoint.insertedSequence = "NA";
+                                if (assembledBreakpoint.region1.endInContig < assembledBreakpoint.region2.startInContig - 1) {
+                                    assembledBreakpoint.insertedSequence = getSequence(contigsCollection, contigId, assembledBreakpoint.region1.endInContig, assembledBreakpoint.region2.startInContig);
+                                }
+
+                                assembledBreakpoints.add(assembledBreakpoint);
+                            }
+                        }
+                    }
+                }
+        } catch (final IOException e) {
+            throw new GATKException("could not execute BWA");
+        }
+
+        return assembledBreakpoints;
+    }
+
+    /**
+     * Pulls the sequence of the requested contig between the start and end coordinates from the contigsCollection
+     * @param contigsCollection
+     * @param contigId
+     * @param startInContig
+     * @param endInContig
+     * @return
+     */
+    private String getSequence(final RunSGAViaProcessBuilderOnSpark.ContigsCollection contigsCollection, final String contigId, final int startInContig, final int endInContig) {
+        for (final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contig : contigsCollection.getContents()) {
+            if (contig._1.equals(new RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID(contigId))) {
+                final String sequence = contig._2.toString();
+                return sequence.substring(startInContig, endInContig);
+            }
+        }
+        throw new GATKException("Could not find contig " + contigId);
```

Comment:
Again, I think index.close() and mem.dispose() should be called before re-throwing.
I think it would be worth it to have a subclass of Exception here, and have an utility function to catch that exception and close C-allocated resources, then re-throw.